content/src/content/jcr_root/apps/core/wcm/components/accordion/v1/accordion/accordion.html

```diff
@@ -13,32 +13,30 @@
   ~ See the License for the specific language governing permissions and
   ~ limitations under the License.
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/-->
-
 <div data-sly-use.accordion="com.adobe.cq.wcm.core.components.models.Accordion"
-     data-sly-use.templates="core/wcm/components/commons/v1/templates.html"
      class="cmp-accordion"
-     data-cmp-is="accordion">
-    <sly data-sly-test="${accordion.items && accordion.items.size > 0}"
-         data-sly-repeat.item="${accordion.items}">
-        <div class="cmp-accordion__item${item.name == accordion.expandedItem ? ' initially-expanded' : ''}"
-             data-cmp-hook-accordion="item"
-             aria-expanded="false"
-             role="button"
-             tabindex="0">
-            <h3 data-sly-element="${accordion.headingType @ context='elementName'}"
-                class="cmp-accordion__item__title"
-                role="heading">
-                ${item.title}
-                <span class="cmp-accordion__item__title__icon"></span>
-            </h3>
-        </div>
-
+     data-cmp-is="accordion"
+     data-cmp-single-expansion="${accordion.singleExpansion}">
+    <div data-sly-test="${accordion.items && accordion.items.size > 0}"
+         data-sly-repeat.item="${accordion.items}"
+         class="cmp-accordion__item"
+         data-cmp-hook-accordion="item"
+         data-cmp-expanded="${item.name in accordion.expandedItems}">
+        <h3 data-sly-element="${accordion.headingElement @ context='elementName'}"
+            class="cmp-accordion__header">
+            <button class="cmp-accordion__button${item.name in accordion.expandedItems ? ' cmp-accordion__button--expanded' : ''}"
+                    aria-expanded="${item.name in accordion.expandedItems ? 'true' : 'false'}"
```

Comment:
I'm still not sure about whether I like this being in the markup. I think this should be added via JS on initialization, so that if JavaScript is disabled on page load, this, along with the `aria-hidden` on panel does not misdirect the user.

This is also not consistent with tabs component, which does not add those attributes until JavaScript initialization: https://github.com/adobe/aem-core-wcm-components/blob/master/content/src/content/jcr_root/apps/core/wcm/components/tabs/v1/tabs/tabs.html

Reply:
But we do add the `.cmp-accordion__panel--expanded` classes, so these panels _will_ be hidden by default (as long as the correct CSS is provided). So the classes `aria` attributes aren't inconsistent with what the user would experience.

Right, we should align with the tabs component.

Reply:
Hmm, I'm not sure if that's right then. Shouldn't users be able to see all of the content if there's no possibility to open the panels? Maybe instead of `aria-hidden` we could add the `hidden` attribute, which will have the same meaning for screen-readers and hide the panels based on that?

Reply:
@jakublamprecht - thanks! I'll try adding this progressive enhancement such that the `aria` attributes and the css classes are applied by the widget (meaning all panels are open in the no-js situation).

Re. the `hidden` attribute I would rather keep the style classes and accessibility attributes separate for clarity.